lxd/api_cluster.go

```diff
@@ -4559,3 +4559,34 @@ func healClusterMember(s *state.State, gateway *cluster.Gateway, op *operations.
 	s.Events.SendLifecycle(api.ProjectDefaultName, lifecycle.ClusterMemberHealed.Event(name, op.EventLifecycleRequestor(), nil))
 	return nil
 }
+
+// clusterGroupUsedBy returns the list of resource URLs that reference the cluster group.
+// The returned slice contains project URLs (for projects whose "restricted.cluster.groups" configuration includes the group) and instance URLs (for instances whose config contains the group in the "volatile.cluster.group" key).
+// If firstOnly is true then search stops at first result.
+func clusterGroupUsedBy(ctx context.Context, s *state.State, tx *db.ClusterTx, name string, firstOnly bool) ([]string, error) {
+	var usedBy []string
+	var err error
+
+	usedBy, err = dbCluster.GetProjectsUsingRestrictedClusterGroups(ctx, tx.Tx(), name)
```

Comment:
if `len(usedBy) >0 && firstOnly` then we should return the first element of `usedBy` here and not proceed to `tx.InstanceList` below.